tests/e2e/stateapp/stateapp_test.go

```diff
@@ -415,3 +481,47 @@ func TestStateTransactionApps(t *testing.T) {
 		})
 	}
 }
+
+func TestMissingStateStore(t *testing.T) {
+	externalURL := tr.Platform.AcquireAppExternalURL(appName)
+	require.NotEmpty(t, externalURL, "external URL must not be empty!")
+
+	// This initial probe makes the test wait a little bit longer when needed,
+	// making this test less flaky due to delays in the deployment.
+	_, err := utils.HTTPGetNTimes(externalURL, numHealthChecks)
+	require.NoError(t, err)
+
+	// Define state URL for statestore that does not exist.
+	stateURL := "http://localhost:3500/v1.0/state/nonexistantstate"
+
+	request := newRequest(utils.SimpleKeyValue{guuid.New().String(), nil})
+	body, _ := json.Marshal(request)
+
+	resp, err := utils.HTTPPost(stateURL, body)
```

Comment:
use PostWithStatus to get the status and verify the status also.

Reply:
Done, though it appears that the state app eats a lot of error messages. You can see it in the response strings though.